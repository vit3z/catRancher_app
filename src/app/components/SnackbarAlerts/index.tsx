
import { Alert, Snackbar } from "@mui/material";
import { SnackbarAlertsProps } from "@/consts/interfaces";
import styles from "./index.module.css";


const SnackbarAlerts:React.FC<SnackbarAlertsProps> = ({
  error,
  turnOffError,
  errorContent,
  warning,
  turnOffWarning,
  success,
  turnOffSuccess,
}) => {
  return (
    <>
      <Snackbar
        open={error}
        autoHideDuration={6000}
        onClose={() => turnOffError(false)}
      >
        <Alert
          onClose={() => turnOffError(false)}
          severity="error"
          variant="filled"
          className={styles.snackBarAlert}
        >
          {`An error has occured: ${errorContent}`}
        </Alert>
      </Snackbar>

      <Snackbar
        open={warning}
        autoHideDuration={6000}
        onClose={() => turnOffWarning(false)}
      >
        <Alert
          onClose={() => turnOffWarning(false)}
          severity="warning"
          variant="filled"
          className={styles.snackBarAlert}
        >
          {`You already have that setup of cats in a Clowder, please switch it up!`}
        </Alert>
      </Snackbar>

      <Snackbar
        open={success}
        autoHideDuration={6000}
        onClose={() => turnOffSuccess(false)}
      >
        <Alert
          onClose={() => turnOffSuccess(false)}
          severity="success"
          variant="filled"
          className={styles.snackBarAlert}
        >
          {`You've successfully created a Clowder!`}
        </Alert>
      </Snackbar>
    </>
  );
};

export default SnackbarAlerts;