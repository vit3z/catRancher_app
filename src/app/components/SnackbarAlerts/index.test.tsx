import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import SnackbarAlerts from './index';

jest.mock('@mui/material', () => ({
  ...jest.requireActual('@mui/material'),
  Snackbar: jest.fn(({ open, autoHideDuration, onClose, children }) => (
    <div className={`mock-snackbar ${open ? 'open' : 'closed'}`}>
      {children}
    </div>
  )),
  Alert: jest.fn(({ onClose, severity, variant, className, children }) => (
    <div className={`mock-alert ${severity} ${variant} ${className}`}>
      {children}
    </div>
  )),
}));

describe('SnackbarAlerts', () => {
  it('renders correctly with error', () => {
    const props = {
      error: true,
      turnOffError: jest.fn(),
      errorContent: 'Sample error message',
      warning: false,
      turnOffWarning: jest.fn(),
      success: false,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles onClose for error', () => {
    const turnOffErrorMock = jest.fn();
    const props = {
      error: true,
      turnOffError: turnOffErrorMock,
      errorContent: 'Sample error message',
      warning: false,
      turnOffWarning: jest.fn(),
      success: false,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const instance = component.root;

    // Simulate closing of the error Snackbar
    instance.findByProps({ severity: 'error' }).props.onClose();

    // Verify if turnOffError was called with the correct arguments
    expect(turnOffErrorMock).toHaveBeenCalledWith(false);
  });

  it('renders correctly with warning', () => {
    const props = {
      error: false,
      turnOffError: jest.fn(),
      errorContent: '',
      warning: true,
      turnOffWarning: jest.fn(),
      success: false,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles onClose for warning', () => {
    const turnOffWarningMock = jest.fn();
    const props = {
      error: false,
      turnOffError: jest.fn(),
      errorContent: '',
      warning: true,
      turnOffWarning: turnOffWarningMock,
      success: false,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const instance = component.root;

    // Simulate closing of the warning Snackbar
    instance.findByProps({ severity: 'warning' }).props.onClose();

    // Verify if turnOffWarning was called with the correct arguments
    expect(turnOffWarningMock).toHaveBeenCalledWith(false);
  });

  it('renders correctly with success', () => {
    const props = {
      error: false,
      turnOffError: jest.fn(),
      errorContent: '',
      warning: false,
      turnOffWarning: jest.fn(),
      success: true,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles onClose for success', () => {
    const turnOffSuccessMock = jest.fn();
    const props = {
      error: false,
      turnOffError: jest.fn(),
      errorContent: '',
      warning: false,
      turnOffWarning: jest.fn(),
      success: true,
      turnOffSuccess: turnOffSuccessMock,
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const instance = component.root;

    instance.findByProps({ severity: 'success' }).props.onClose();
    expect(turnOffSuccessMock).toHaveBeenCalledWith(false);
  });

  it('renders correctly with all types of alerts closed', () => {
    const props = {
      error: false,
      turnOffError: jest.fn(),
      errorContent: '',
      warning: false,
      turnOffWarning: jest.fn(),
      success: false,
      turnOffSuccess: jest.fn(),
    };
    const component = renderer.create(<SnackbarAlerts {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
