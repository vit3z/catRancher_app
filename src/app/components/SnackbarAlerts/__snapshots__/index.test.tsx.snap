// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarAlerts renders correctly with all types of alerts closed 1`] = `
[
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert error filled snackBarAlert"
    >
      An error has occured: 
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert warning filled snackBarAlert"
    >
      You already have that setup of cats in a Clowder, please switch it up!
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert success filled snackBarAlert"
    >
      You've successfully created a Clowder!
    </div>
  </div>,
]
`;

exports[`SnackbarAlerts renders correctly with error 1`] = `
[
  <div
    className="mock-snackbar open"
  >
    <div
      className="mock-alert error filled snackBarAlert"
    >
      An error has occured: Sample error message
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert warning filled snackBarAlert"
    >
      You already have that setup of cats in a Clowder, please switch it up!
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert success filled snackBarAlert"
    >
      You've successfully created a Clowder!
    </div>
  </div>,
]
`;

exports[`SnackbarAlerts renders correctly with success 1`] = `
[
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert error filled snackBarAlert"
    >
      An error has occured: 
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert warning filled snackBarAlert"
    >
      You already have that setup of cats in a Clowder, please switch it up!
    </div>
  </div>,
  <div
    className="mock-snackbar open"
  >
    <div
      className="mock-alert success filled snackBarAlert"
    >
      You've successfully created a Clowder!
    </div>
  </div>,
]
`;

exports[`SnackbarAlerts renders correctly with warning 1`] = `
[
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert error filled snackBarAlert"
    >
      An error has occured: 
    </div>
  </div>,
  <div
    className="mock-snackbar open"
  >
    <div
      className="mock-alert warning filled snackBarAlert"
    >
      You already have that setup of cats in a Clowder, please switch it up!
    </div>
  </div>,
  <div
    className="mock-snackbar closed"
  >
    <div
      className="mock-alert success filled snackBarAlert"
    >
      You've successfully created a Clowder!
    </div>
  </div>,
]
`;
